scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -409,6 +481,8 @@ object Mod {
   @ast class ValParam() extends Mod
   @ast class VarParam() extends Mod
   @ast class Inline() extends Mod
+  @ast class Using() extends Mod
```

Comment:
Why would `Using()` be a modifier?

Reply:
It behaves a modifier `def f(x: String)(using Int)(y: String)(using b: Int)`
Do you think it would be better to represent it in other way?

Reply:
It is kind of a modifier, but it applies to the whole parameter *list*, not just one parameter (unlike the other modifiers above). In dotty itself it's a flag of the parameter list, not a modifier of each parameter.

Reply:
Yeah well this is a little messy and I am not sure how to make it properly.
There are 2 situations, definition and usage.
For method definition we store `paramss: List[List[Term.Param]]`. I can't really do much here without bigger changes, I decided to add `Using` modifier to each param that is in a list if `using` modifier was found in this list. It was nice during parsing, but now I hit terrible problems with printing this structure(no idea how to do it yet)
On method call it is much better, because we have `Apply(term, List[Term]`, I just added `ApplyUsing(term, List[Term])` and it covers it nicely

I can imagine having `sparams: List[List[Term.Param]]` changed to
`sparams: List[TreeList[Term.Param]]` where
`TermList <: TreeList` and `TermListUsing <: TreeList` but I don't think it is doable in reasonable effort not talking about compatibility :/

Reply:
Ok actually it is similar to `implicit` we have in scala2, I will take inspiration from there :P

Reply:
reworked it to keep it the same way as implicit.